src/download-file.js

```diff
@@ -19,6 +21,11 @@ const downloadFile = (file) => {
   return new Promise((resolve, reject) => {
     fs.stat(fileName, (err) => {
       if (err) { // File doesn't exists yet
+        const { dir } = parsePath(fileName);
+
+        // TODO: Change synchronous call.
+        fse.ensureDirSync(dir);
```

Comment:
I'm thinking how to do this. The problem here is the `new Promise`. Since it receives a function, we can't apply async/await here. I think that it's possible if the `request` block is moved to a different function, separating it from the `fs.stat`. In that way, `fs.stat` can use async/await too.